### ISSUE TITLE
bruteforce incorrect ulp error report for cos(double)

### ISSUE THREAD

**Issue by b-sumner** [KhronosGroup/OpenCL-CTS#1329]
Running 'bruteforce -rf1 cos' on my implementation produces:

  1:             cos fp64     ................passed        2.27 @ 0x1.d10000000074p+380

Now, for that particular input argument, my implementation produces:

  0.46 ulp at (0x1.d100000000740p+380): *0x1.ffff1d06dd23f8a8p-8 vs 0x1.ffff1d06dd240p-8

Wolfram claims the correct answer for this argument is
0.0078124471536865077073474543921834934716746213988958700965071438...

Which in hex rounds to: 0x1.ffff1d06dd240p-8 in double precision and identical to what my implementation produces.

So why is bruteforce claiming my implementation has a 2.27 ulp error for this input?

**Comment by bashbaug** [comment]
I'm not exactly sure where to go with this, but I do observe that the reference value for this input is:

```
cos(0x1.d10000000074p+380) -> 0x1.ffff1d06dd23ep-8 (0.00781245) (as double)
   -> 0xf.fff8e836e91eddcp-11 (0.00781245) (as long double)
```

This was computed by adding to the top of the bruteforce main.cpp:

```c
    double d = 0x1.d10000000074p+380;
    double r = reference_cos(d);
    long double l = reference_cosl(d);
    printf("cos(%a) -> %a (%g) (as double)\n   -> %La (%Lg) (as long double)\n", d, r, r, l, l);
```

I agree this seems a bit incorrect.  The results I get from Sollya are:

```
> cos(0x1.d10000000074p+380);
Warning: rounding has happened. The value displayed is a faithful rounding to 165 bits of the true result.
0x1.ffff1d06dd23f8a744c91c39edb08c52ca459d1fp-8
```

Do we need to look at the reference cosine calculation?  Or did I do something wrong here?

**Comment by b-sumner** [comment]
I don't think you did anything wrong.  I think reference_cosl() is not producing a good result for this input (at least).
